### PR TITLE
Enhancement

### PR DESCRIPTION
**Pull request by wongcyrus**
One click ARM deployment.
Change to use Azure Table.
Add student registration.
Add Student Registration demo video and registration steps in README.
Update GraderArchitecture.png with Azure Table.

### CODE REVIEW

**Review by leestott — CHANGES_REQUESTED** [review]
AzureAutomaticGradingEngineFunctionApp/StudentRegistrationFunction.cs
                log.LogInformation("Form Submit");
                string project = req.Form["project"];
                string email = req.Form["email"];
                string credentials = req.Form["credentials"];
                log.LogInformation("Student Register: " + email + " Project:" + project);
                if (string.IsNullOrWhiteSpace(project) || string.IsNullOrWhiteSpace(email) || string.IsNullOrWhiteSpace(credentials))
                {
Step 2 access to local variable credentials
Sink
AzureAutomaticGradingEngineFunctionApp/StudentRegistrationFunction.cs
                CloudTable credentialsTable = tblclient.GetTableReference("credentials");
                CloudTable subscriptionTable = tblclient.GetTableReference("subscriptions");

                var credentialObject = Newtonsoft.Json.JsonConvert.DeserializeObject<dynamic>(credentials);
 flows to unsafe deserializer.
                string subscriptionId = credentialObject.subscriptionId;

                TableResult result = await subscriptionTable.ExecuteAsync(TableOperation.Retrieve(project, subscriptionId, new List<string>() { "Email" }));

File: azuredeploy.json [hunk 1]
```diff
@@ -17,34 +17,119 @@
...
+              "name": "AzureWebJobsStorage",
+              "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('uniqueName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('uniqueName')), '2019-06-01').keys[0].value,';EndpointSuffix=','core.windows.net')]"
+            },
+            {
+              "name": "storageTestResult",
+              "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('uniqueName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('uniqueName')), '2019-06-01').keys[0].value,';EndpointSuffix=','core.windows.net')]"
+            },
+            {
+              "name": "WEBSITE_RUN_FROM_PACKAGE",
+              "value": "https://github.com/microsoft/AzureAutomaticGradingEngine/raw/master/Prebuilt/AzureAutomaticGradingEngineFunctionApp.zip"
```
Comment (MaximRouiller) [inline azuredeploy.json:61]: Do we have a `master` branch?

File: README.md [hunk 2]
```diff
@@ -16,22 +16,16 @@ The project is being validated through usage on the course [Higher Diploma in Cl
 You can read more about this project at [Microsoft Educator Developer TechCommunity](https://techcommunity.microsoft.com/t5/educator-developer-blog/microsoft-azure-automatic-grading-engine/ba-p/2681809?WT.mc_id=academic-39457-leestott)
 ## Prerequisite
 
-- 1 Storage account with 2 containers
-- testresult and credentials with resource group name "azureautomaticgradingengine".
-
 [![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2FAzureAutomaticGradingEngine%2Fmaster%2Fazuredeploy.json)
```
Comment (MaximRouiller) [inline README.md:19]: Do we have a `master` branch?

File: azuredeploy.json (hunk 1 above)
Comment (leestott) [reply azuredeploy.json:61]: good catch.

**Review by leestott — APPROVED** [review]

**Review by leestott — APPROVED** [review]
I will update the web app deployment to main after PR merge

**Pull request merged**